src/components/profile/EditProfile.js

```diff
@@ -37,79 +35,70 @@ const EditProfile = ({ screenProps, styles, navigation }) => {
   const [showErrorDialog] = useErrorDialog()
   const { push } = screenProps
 
-  useEffect(() => {
-    if (isEqual(storedProfile, {})) {
-      updateProfile()
-    }
-  }, [])
-
-  // Validate after saving profile state in order to show errors
-  useEffect(() => {
-    //need to pass parameters into memoized debounced method otherwise setX hooks wont work
-    validate()
-  }, [profile])
+  const deboucedProfile = useDebounce(profile, 500)
+  const onProfileSaved = useCallback(() => push(`Dashboard`), [push])
+  const handleEditAvatar = useOnPress(() => push(`ViewAvatar`), [push])
 
   const updateProfile = useCallback(async () => {
     // initialize profile value for first time from storedProfile in userStorage
     const profileFromUserStorage = await userStorage.getProfile()
-    store.set('privateProfile')(profileFromUserStorage)
+
+    setPrivateProfile(profileFromUserStorage)
     setProfile(profileFromUserStorage)
-  }, [store, setProfile])
-
-  const validate = useCallback(
-    debounce(async () => {
```

Comment:
its very easy to verify if this was working or not. only replace if you saw it is not working.

Reply:
There was left a comment that it’s not working, we have to pass args to debounced function. I googled about that and found that’s right, a lot of people have met with that issue. And then I found a custom hook solving the issue in more elegant way.

About debouce issue:
https://stackoverflow.com/questions/61785903/problems-with-debounce-in-useeffect
https://stackoverflow.com/questions/57547582/useeffect-with-debounce
Report about the issue in the comment:
https://github.com/GoodDollar/GoodDAPP/blob/39fb123bef3dd68b45e3b9502c2a31b1e10f7bf7/src/components/profile/EditProfile.js#L48